### PR TITLE
Update CONTRIBUTING.md for clarity and readability

### PR DESCRIPTION
**Pull request by acicovic**
## Description
This PR started as a quick fix for a few CONTRIBUTING.md phrase improvement suggestions that were made in #608. However, it has evolved to a small restructure of the CONTRIBUTING.md file, as well as a minimal rewrite of some sections so they can make more sense between them.

## Motivation and Context
As the newest contributor in the team and my relative inexperience with all the tooling, I've tried to update CONTRIBUTING.md in a way that is more easy to undertand and follow.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,16 +1,38 @@
...
+
+### Important branches
+
+Ongoing development is being done in the `develop` branch. Merges are performed against the `trunk` branch once considered stable.
+
+To contribute code to this project, fork the repo and open a PR to the `develop` branch. Alternatively, if you have push access to this repo, create a feature branch and then open an intra-repo PR from that branch to `develop`.
+
+### Coding standards
+
+The Parse.ly plugin uses the PHP_CodeSniffer tool that is installed through Composer. This plugin uses a [custom ruleset.](https://github.com/Parsely/wp-parsely/blob/develop/.phpcs.xml.dist)
```
Comment (GaryJones) [inline CONTRIBUTING.md]:
```suggestion
The Parse.ly plugin uses the PHP_CodeSniffer tool that is installed through Composer. This plugin uses a [custom ruleset](https://github.com/Parsely/wp-parsely/blob/develop/.phpcs.xml.dist).
```

**Review by GaryJones — DISMISSED** [review]
One tiny fix, then all good.

File: CONTRIBUTING.md (hunk 1 above)
Comment (acicovic) [reply CONTRIBUTING.md]: Nice catch.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,16 +1,38 @@
...
 
 Please search the [repo's issues](https://github.com/Parsely/wp-parsely/issues) to see if your issue has been reported already and if so, comment on that issue instead of opening a new one.
 
 When creating a new issue, please add specific steps to reproduce the problem, upload any relevant screenshots, and describe what happened and what you expected would happen instead.
 
-## Setting up your environment
+## Contributing code
 
-### Minimum required versions
+You are more than welcome to submit PRs that fix issues or introduce new features to the Parse.ly plugin.
```
Comment (pauarge) [inline CONTRIBUTING.md]:
```suggestion
You are welcome to contribute to the plugin by submitting Pull Requests fix issues or introduce new features to the Parse.ly plugin.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,16 +1,38 @@
...
+
+Ongoing development is being done in the `develop` branch. Merges are performed against the `trunk` branch once considered stable.
+
+To contribute code to this project, fork the repo and open a PR to the `develop` branch. Alternatively, if you have push access to this repo, create a feature branch and then open an intra-repo PR from that branch to `develop`.
+
+### Coding standards
+
+The Parse.ly plugin uses the PHP_CodeSniffer tool that is installed through Composer. This plugin uses a [custom ruleset](https://github.com/Parsely/wp-parsely/blob/develop/.phpcs.xml.dist).
+
+The plugin aims to use strong types where possible, so be sure to declare `strict_types=1` on new files, and include type definitions for parameters and return types that are compatible with the minimum version of PHP that this plugin supports.
```
Comment (pauarge) [inline CONTRIBUTING.md]:
```suggestion
The plugin uses strong types, so be sure to declare `strict_types=1` on new files, and include type definitions for parameters and return types that are compatible with the minimum version of PHP that this plugin supports.
```

**Review by pauarge — CHANGES_REQUESTED** [review]
Left some small suggestions

**Comment by acicovic** [comment]
The masterpiece is complete (for now).

**Review by pauarge — APPROVED** [review]
LGTM

**Pull request merged**